src/core/ext/filters/client_channel/xds/xds_client.cc

```diff
@@ -756,8 +837,30 @@ void XdsClient::ChannelState::AdsCallState::SendMessageLocked(
   }
 }
 
+void XdsClient::ChannelState::AdsCallState::Subscribe(
+    const std::string& type_url, const std::string& name) {
+  auto& state = state_map_[type_url].subscribed_resources[name];
+  if (state == nullptr) {
+    state = MakeOrphanable<ResourceState>(type_url, name);
+    SendMessageLocked(type_url);
+  }
+}
+
+void XdsClient::ChannelState::AdsCallState::Unsubscribe(
+    const std::string& type_url, const std::string& name) {
+  state_map_[type_url].subscribed_resources.erase(name);
+  SendMessageLocked(type_url);
```

Comment:
should this line be called before the previous line?  The caller of this method was doing the send first and then the remove